peps/pep-0735.rst

```diff
@@ -0,0 +1,476 @@
+PEP: 735
+Title: Dependency Groups in pyproject.toml
+Author: Stephen Rosen <sirosen0@gmail.com>
+Sponsor: Brett Cannon <brett@python.org>
+Discussions-To: https://discuss.python.org/t/39233
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 20-Nov-2023
+Post-History: `20-Nov-2023 <https://discuss.python.org/t/39233>`__, `14-Nov-2023 <https://discuss.python.org/t/29684>`__
+
+Abstract
+========
+
+This PEP specifies a mechanism for storing package requirements in
+pyproject.toml files such that it is not included in any built distribution of
+the project.
+
+This is suitable for creating named groups of dependencies, similar to
+``requirements.txt`` files, which launchers, IDEs, and other tools can find and
+identify by name.
+
+The feature defined here is referred to as "Dependency Groups".
+
+Motivation
+==========
+
+There are two major use-cases for which the Python community has no
+standardized answer:
+
+* How should development dependencies be defined for packages?
+
+* How should dependencies be defined for projects which do not build
+  distributions (non-package projects)?
+
+In the absence of any standard, two known workflow tools, PDM and Poetry, have
+defined their own solutions for Dependency Groups to address the first of these
+two needs. Their definitions are very similar to one another, although PDM
+structures them much more similarly to extras than Poetry does.
```

Comment:
I've linked to the section in Dependency Specifiers ( https://packaging.python.org/en/latest/specifications/dependency-specifiers/ ) which describes extras. I just noticed that this might not be the right place to point. Should I use https://packaging.python.org/en/latest/specifications/core-metadata/#provides-extra-multiple-use ? Or maybe there's some better place to link? I really want to point at the specification for `project.optional-dependencies` but I didn't find it easily.